pkg/controller/routestate/route_state_controller.go

```diff
@@ -0,0 +1,246 @@
+package routestate
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const routeStateControllerWorkQueueKey = "key"
+
+type RouteStateController struct {
+	endpointsGetter corev1client.EndpointsGetter
+	podGetter       corev1client.PodsGetter
+	queue           workqueue.RateLimitingInterface
+	cachesToSync    []cache.InformerSynced
+	targetNamespace string
+	operatorClient  v1helpers.OperatorClient
+	eventRecorder   events.Recorder
+}
+
+func NewRouteStateController(kubeInformersForTargetNamespace informers.SharedInformerFactory,
```

Comment:
We're checking service endpoints here, isn't this really but a NetworkStateController?

Reply:
no, but I think IngressStateController might be more accurate, renamed.